packages/application-shell/src/apollo-links/token-retry-link.ts

```diff
@@ -18,34 +20,48 @@ type ApolloContext = {
 // so that the MC BE can issue a new token.
 // NOTE: the retry is not meant to work for the MC access token.
 
+const TOKEN_RETRY_HEADER_NAME = 'X-Force-Token';
+
 export const getDoesGraphQLTargetSupportTokenRetry = (
   requestHeaders: ApolloContext['headers']
-) => {
+): boolean => {
   const target = (requestHeaders['X-Graphql-Target'] ||
     requestHeaders['x-graphql-target']) as TokenRetryGraphQlTarget;
+
   return [
     GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
     GRAPHQL_TARGETS.ADMINISTRATION_SERVICE,
     GRAPHQL_TARGETS.SETTINGS_SERVICE,
     GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND,
   ].includes(target);
 };
+export const getSkipTokenRetry = (
+  variables: Operation['variables']
+): boolean => {
+  const skipTokenRetry = Boolean(variables.skipTokenRetry);
+
+  return skipTokenRetry;
+};
+
+const forwardTokenRetryHeader = (
```

Comment:
Extracted. While doing I noticed that the `true` as boolean is not complying with `Record<string, string>` so I changed it.